File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        s_counter = Counter(s)
        res = 0
        for char in s_counter:
            res += s_counter[char] // 2 * 2
            if res % 2 == 0 and s_counter[char] % 2 == 1:
                res += 1
        return res

File: 0409-longest-palindrome/0409-longest-palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    let odd = false
    let chars = {}
    for (let char of s) {
        if (!chars.hasOwnProperty(char)) {
            chars[char] = 0
        }
        chars[char] += 1
    }
    let res = 0
    for (let char in chars) {
        if (chars[char] % 2 === 1) odd = true
        res += ~~(chars[char] / 2) * 2
    }
    return odd ? res + 1 : res
};